pol-core/pol/decay.cpp

```diff
@@ -116,36 +116,33 @@ void decay_worldzone( unsigned wx, unsigned wy, Realms::Realm* realm )
 ///     once every 10 minutes
 ///
 
-void decay_single_zone( Realms::Realm* realm, unsigned gridx, unsigned gridy, unsigned& wx,
-                        unsigned& wy )
+void decay_single_zone( Realms::Realm* realm, const Pos2d& gridsize, Pos2d& pos )
```

Comment:
I wonder if this function could be replaced by an iterator on the range in the other loop, because it's basically doing two things:
- iterate the range
- decay the zone

Might be better to iterate the range directly in the other loop. Didn't you make a range2d or so?

Reply:
Created issue #653